### ISSUE TITLE
Question: Advice on training neural ODE for sigmoid type function

### ISSUE THREAD

**Issue by TysonStrath** [patrick-kidger/diffrax#677]
Hi everyone, 

Thanks for the great package Patrick. I am quite new to neural ODEs and was wondering if anyone had advice on training the neural ODE in this package for sigmoid functions. I set up the vector field as a coupled ODE between the neural ODE and a linear ODE (heating rate). I have already implemented a weighted loss function around the data median, but it struggles to learn the curve. 

This is the output:
<img width="589" height="606" alt="Image" src="https://github.com/user-attachments/assets/52b07153-3ac9-4132-9885-e68b2153ce22" />

This is my ODE initialization: 
```
class NeuralODE(eqx.Module):
    out_scale: jax.Array
    mlp: eqx.nn.MLP
    def __init__(self, data_size, width_size, depth, *, key, **kwargs):
        super().__init__(**kwargs)
        self.out_scale = jnp.array(1.0)
        self.mlp = eqx.nn.MLP(
            in_size=2, #takes in temperature and concentration
            out_size=1, #this needs to be only concentration
            width_size=width_size,
            depth=depth,
            activation=jax.nn.softplus,
            use_final_bias=False,
            key=key,
        )
    def __call__(self, t, y, args=None):
        temp7, c7 = y
        input = jnp.array([temp7, c7])
        return self.mlp(input) * self.out_scale

class CoupledODE(eqx.Module):
    nn: NeuralODE
    heating_rate: float = eqx.field(static=True)
    def __call__(self, t, y, args):
      dTdt = self.heating_rate
      dcdt = self.nn(t, y)
      dcdt_scalar = jnp.squeeze(dcdt)
      dydt = jnp.stack([dTdt, dcdt_scalar])
      return dydt
```

Thanks, 
Tyson

**Comment by patrick-kidger** [comment]
Hey there!
The first thing that jumps out is that you don't have a squashing final activation. I actually did recently update the neural ODE example to do the correct thing on this point.
Learning over increasing-length time intervals is another standard trick.
Plus: playing with learning rates etc. :)

**Comment by TysonStrath** [comment]
Thanks Patrick!
